src/unk_02092BE8.c

```diff
@@ -0,0 +1,288 @@
+#include "unk_02092BE8.h"
+#include "safari_zone.h"
+#include "constants/sndseq.h"
+
+void sub_02092BE8(FieldSystem* sys, UnkSTRUCT_2C* ptr, BOOL Unkbool) { //r5, r4, SP
+    UnkSTRUCT_14 Unk_struct;
+    SCRIPT_STATE *state = SavArray_Flags_get(sys->savedata);
+    FLYPOINTS_SAVE *points = Save_FlyPoints_get(sys->savedata); 
+    Location *warpPtr = FlyPoints_GetSpecialSpawnWarpPtr(points); //r6
+    Location *PosPtr = FlyPoints_GetPosition(points); //SP + 4
+    UnkFsysSub_114* Unk_V1 = sub_02092DEC(sys); //r7
+    ptr->data = sys->savedata;
+    ptr->mapID = PosPtr->mapId;
+    if(MapHeader_MapIsOnMainMatrix(ptr->mapID) != 0){ 
```

Comment:
```suggestion
    if (MapHeader_MapIsOnMainMatrix(ptr->mapID) != 0) { 
```

other similar style changes in this file